routers/web/admin/emails.go

```diff
@@ -150,3 +151,33 @@ func ActivateEmail(ctx *context.Context) {
 	redirect.RawQuery = q.Encode()
 	ctx.Redirect(redirect.String())
 }
+
+// DeleteEmail serves a POST request for delete a user's email
+func DeleteEmail(ctx *context.Context) {
+	u, err := user_model.GetUserByID(ctx, ctx.FormInt64("Uid"))
+	if err != nil || u == nil {
+		ctx.ServerError("GetUserByID", err)
+		return
+	}
+
+	email, err := user_model.GetEmailAddressByID(ctx, u.ID, ctx.FormInt64("id"))
+	if err != nil || email == nil {
+		ctx.ServerError("GetEmailAddressByID", err)
+		return
+	}
+
+	redirect := setting.AppSubURL + "/admin/emails"
+	if err := user.DeleteEmailAddresses(ctx, u, []string{email.Email}); err != nil {
+		if user_model.IsErrPrimaryEmailCannotDelete(err) {
+			ctx.Flash.Error(ctx.Tr("admin.emails.delete_primary_email_error"))
+			ctx.JSONRedirect(redirect)
```

Comment:
It shouldn't redirect to `/admin/emails`. Think about a case: the admin is on the 2nd page, then the error occurs, then they gets redirected to the first page incorrectly.

Reply:
What is `the 2nd page`?

Reply:
Emmm, query will be ignored, maybe this is a problem. I will fix it.

Reply:
I think 			`ctx.JSONRedirect("")` is enough, it reloads current page.

Reply:
Done in 73a9bc559cafd096707b05d6c9d4b8403b5a9c97